akka-stream/src/main/scala/akka/stream/javadsl/Flow.scala

```diff
@@ -1485,6 +1485,27 @@ final class Flow[-In, +Out, +Mat](delegate: scaladsl.Flow[In, Out, Mat]) extends
   def detach: javadsl.Flow[In, Out, Mat] = new Flow(delegate.detach)
 
   /**
+   * Materializes to `Future[Unit]` that completes on getting termination message.
+   * The Future completes with success when received complete message from upstream or cancel
+   * from downstream. It fails with the same error when received error message from
+   * downstream.
+   *
+   * `watchTermination` ignores the current [[Flow]] materialization value, in other words it's combine
+   * values with Keep.right
+   */
+  def watchTermination[M](): javadsl.Flow[In, Out, Future[Unit]] =
+    new Flow(delegate.watchTermination())
+
+  /**
+   * Materializes to `Future[Unit]` that completes on getting termination message.
+   * The Future completes with success when received complete message from upstream or cancel
+   * from downstream. It fails with the same error when received error message from
+   * downstream.
+   */
+  def watchTerminationMat[M]()(matF: function.Function2[Mat, Future[Unit], M]): javadsl.Flow[In, Out, M] =
```

Comment:
I’d prefer that the method be called `watchTermination`—since it must deal with the materialized value anyway.